### ISSUE TITLE
Rename master branch to main

### ISSUE THREAD

**Issue by pebrc** [elastic/cloud-on-k8s#5017]
We would like to rename the current `master` branch to `main`. 

Timeline: right after the next (1.9) release (just to avoid any rename related hiccups impacting our ability to ship a release)

Things to consider: 
* Github has made it [much easier to rename branches](https://github.com/github/renaming#rename-existing)
* We are referencing `master` in our CI jobs, there is an `e2e/master` configuration for example, many if not all jobs except the release build are referencing the `master` branch
* Our docs might reference `master` but I think that is actually a bug and should be replaced with a variable pointing to the most recent release or the current branch  (so no hard-coding of the branch name)

Steps:

- [x] Plan the rename, flesh out the points I mentioned above, decide on the sequence of rename steps (when: can start now)
- [x] Rename according to the plan from the previous step (when: right after 1.9)

**Comment by thbkrkr** [comment]
### (draft) Plan

#### Day before the renaming
- [ ] prepare elastic/cloud-on-k8s PR **A** to rename all occurences of `master` to `main` (see script below))
- [ ] prepare elastic/docs PR **B** to update the docs build config to point to main ([example](https://github.com/elastic/docs/pull/2266/files))

#### Day of the renaming
- [ ] rename `master` branch to `main` via the [GitHub UI](https://github.com/elastic/cloud-on-k8s/settings/branches) 
- [ ] update the pattern of the `master` branch protection rule via the [GitHub UI](https://github.com/elastic/cloud-on-k8s/settings/branches) 
- [ ] merge PR **A** & PR **B**
- [ ] send an email to the team to warn of the branch renaming and explain how to update their local clone


-----------

Script to help with the branch renaming for PR **A**:

```sh
#!/bin/bash -eux

replace() { xargs -n1 sed -i "s|$1|$2|g"; }

# rename files containing master
mv .ci/jobs/e2e-tests-master-gke.yml              .ci/jobs/e2e-tests-main-gke.yml
mv .ci/pipelines/e2e-tests-master-gke.Jenkinsfile .ci/pipelines/e2e-tests-main-gke.Jenkinsfile

# rename all occurences of `master` to `main` in `.ci`
grep master -lr .ci  | replace master main

# rename all occurences of `cloud-on-k8s/(blob|tree)/master/` to `cloud-on-k8s/(blob|tree)/main/` in the whole repo
grep master -lr .    | replace cloud-on-k8s/blob/master/  cloud-on-k8s/blob/main/
grep master -lr .    | replace cloud-on-k8s/tree/master/  cloud-on-k8s/tree/main/

**Comment by thbkrkr** [comment]
> Our docs might reference master but I think that is actually a bug and should be replaced with a variable pointing to the most recent release or the current branch.

Yes, we have doc links of ECK and also ES and Beats that reference `master` in a few places.
```
> grep 'https://www.elastic.co.*master[^[#.]*' -ro | sed "s|/.*[doctmpl]:|:|" | sort | uniq -c | sort -rn
     43 docs:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}
     24 docs:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{parent_page_id}
      3 config:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-virtual-memory
      2 hack:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}
      2 docs:https://www.elastic.co/guide/en/elasticsearch/plugins/master/repository-gcs
      1 pkg:https://www.elastic.co/guide/en/beats/filebeat/master/configuration-autodiscover-hints
      1 docs:https://www.elastic.co/guide/en/elasticsearch/reference/master/autoscaling-apis
      1 docs:https://www.elastic.co/guide/en/elasticsearch/plugins/master/repository-gcs-usage
      1 docs:https://www.elastic.co/guide/en/cloud-on-k8s/master/index
      1 docs:https://www.elastic.co/guide/en/beats/filebeat/master/running-on-kubernetes
      1 config:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-operator-config
```

The links to the ES and beats doc and the links in config/ can be updated with `current` (PR TODO). The others are a bit special as they are used for the link behind the text `View this document on the Elastic website` only displayed on GitHub (ex: https://github.com/elastic/cloud-on-k8s/blob/master/docs/quickstart.asciidoc):
```
ifdef::env-github[]
****
link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{parent_page_id}.html#k8s-{page_id}[View this document on the Elastic website]
****
endif::[]
```
In this context, I think `master|main` makes sense. However, there is a bug when looking at a doc page from another branch than `master` (ex: https://github.com/elastic/cloud-on-k8s/blob/1.8/docs/quickstart.asciidoc). Ideally we would want the current GitHub branch but I don't know a directive for this.

We might choose to remove this feature because it's just a nice-to-have non-essential thing that other Elastic products don't have.

If we keep it, we need to update links with `main` only after the Elastic docs is published with URLs containing `main`, which will happen once all content repos have switched to using a `main` branch. In the meantime, the docs continue to be generated under `master` for all repositories (those who have switched and those who have not).

**Comment by barkbay** [comment]
> We might choose to remove this feature because it's just a nice-to-have non-essential thing that other Elastic products don't have.

An other option would be to not set a branch here, and redirect the user to https://www.elastic.co/guide/en/cloud-on-k8s/index.html:

```asciidoc
ifdef::env-github[]
****
link:https://www.elastic.co/guide/en/cloud-on-k8s/index.html[View the documentation on the Elastic website]
****
endif::[]
```

The drawback is that it would require for the user to select the expected version of the documentation, and then navigate to the expected page. It would still be a way to state that this page is not intended to be displayed on GitHub though.

**Comment by thbkrkr** [comment]
> The drawback is that it would require for the user to select the expected version of the documentation, and then navigate to the expected page. It would still be a way to state that this page is not intended to be displayed on GitHub though.

Yes, this may be a reasonable alternative. My first thought was that this drawback makes the feature useless. I had forgotten that the original intention was not to provide a quick way to view the rendered documentation page but rather to redirect GitHub documentation viewers to the Elastic website as GitHub does not support ASCIIDoctor (#1894).

**Comment by thbkrkr** [comment]
Closing as all the work is done. I opened #5338 for the last task to do after the Elastic documentation uses `main` in the URLs.
